### ISSUE TITLE
vmagent mixes metric names, labels and values

### ISSUE THREAD

**Issue by ptimofee** [VictoriaMetrics/VictoriaMetrics#4961]
### Describe the bug

vmagent mixes metrics names, labels and values after upgrade to v1.93.1.
Some random metrics get  random label names as well as values. It looks like all metrics coming through vmagent are now randomly mixed.
This bug is a consequence of https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4932

### To Reproduce

Our config is pretty much simple, we have 3 vmagents within k8s cluster running as shippers: i. e. they scrape nothing, but accept remote_writes from a bunch of prometheus, add some labels and route the data to a few endpoints via remote_write as well.

config file:
```
global:
  scrape_interval: 15s
  external_labels:
    prometheus: monitoring/shipper
scrape_configs: []
```

arg list:
```
    - -http.pathPrefix=/vmagent
    - -httpListenAddr=:8429
    - -promscrape.config=/etc/vmagent/config_out/vmagent.env.yaml
    - -remoteWrite.basicAuth.passwordFile=/etc/vmagent/config/RWS_0-SECRET-BASICAUTHPASSWORD,,,
    - -remoteWrite.basicAuth.username="<FILTERED_USER>","","",""
    - -remoteWrite.label=environment=prod,tenant=<FILTERED_TENANT>
    - -remoteWrite.maxDiskUsagePerURL=21474836480
    - -remoteWrite.showURL=true
    - -remoteWrite.tlsCAFile=/etc/vmagent-tls/certs/our-ca_ca,,,
    - -remoteWrite.tmpDataPath=/vmagent_pq/vmagent-remotewrite-data
    - -remoteWrite.url=https://<ENDPOINT_1>/vmdata/api/v1/write,http://<ENDPOINT_2>/api/v1/write,http://<ENDPOINT_3>/api/v1/write,http://<ENDPOINT_4>/api/v1/write
```

### Version

1.93.1

### Logs

Nothing really to show here

### Additional information

This [BUGFIX](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/v1.93.1-cluster/docs/CHANGELOG.md?plain=1#L31) looks suspicious

**Comment by Amper** [comment]
Hey @ptimofee.

I couldn't reproduce this bug on `v1.93.1` and `1.93.3`.

I used following configuration:

![image](https://github.com/VictoriaMetrics/VictoriaMetrics/assets/495795/0a836d54-d295-49b7-8316-d07e80057cb6)

After my tests all data looked expected and were completely consistent across all vmsingles.

It's possible it depends on something else:
- Specific sent data
- Amount of data and load on vmagent
- Some event, errors in the logs, etc.

Could you please provide more details on exactly what the results of the bug look like and when, how and under what circumstances it is reproduced?

**Comment by ptimofee** [comment]
I've been busy with other stuff, today I'm going to focus on reproduction of the issue. I'll provide my finding.

**Comment by ptimofee** [comment]
I can't reproduce in a lab like you built as well.
But I could do that multiple times in our prod. Maybe there are much more metrics or something else going on there.

So I can confirm the issue is in 1.93.1+ as I can't reproduce it with 1.93.0.

Digging further to see what option or knob triggers it in prod to try it in lab

**Comment by ptimofee** [comment]
So far so good: if I remove -remoteWrite.label then 1.93.1 behaves ok.

**Comment by Amper** [comment]
Then it looks like a bug made in [this issue](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4247). 

@ptimofee thanks a lot, i'll figure it out and return with feedback or buffix.

**Comment by f41gh7** [comment]
Hello, a bug must be fixed at the related PR. It's added at the v1.93.3 version during fixes for this issue https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4247

**Comment by ptimofee** [comment]
@f41gh7 @Amper ok, building and trying the fix from https://github.com/VictoriaMetrics/VictoriaMetrics/pull/4983

**Comment by ptimofee** [comment]
Running this build for 5 hours almost. No issues so far.
I'll leave it for the weekend.

**Comment by valyala** [comment]
The related issues https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972 and https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4982

**Comment by valyala** [comment]
Closing this issue as duplicate for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972 . See more details at https://github.com/VictoriaMetrics/VictoriaMetrics/issues/4972#issuecomment-1712412380

**Comment by valyala** [comment]
FYI, the bugfix for this issue has been included in [v1.93.4](https://docs.victoriametrics.com/CHANGELOG.html).
